r/vignettes/developing.Rmd

```diff
@@ -11,8 +11,8 @@ If you're interested in contributing to arrow, this vignette explains our approa
 at a high-level.  If you're looking for more detailed content, you may want to 
 look at one of the following links:
 
-* [setting up a development environment and building the components that make up the Arrow project and R package](https://arrow.apache.org/docs/r/articles/developers/setup.html)
-* [common Arrow dev workflow tasks](https://arrow.apache.org/docs/r/articles/developers/workflow.html)
+* [setting up a development environment and building the components that make up the Arrow project and R package](https://arrow.apache.org/docs/r/articles/developing.html#developer-environment-setup)
+* [common Arrow dev workflow tasks](https://arrow.apache.org/docs/r/articles/developing.html#common-developer-workflow-tasks)
```

Comment:
This will be fine for the CRAN checks, but once the pkgdown website is updated, these URLs will just point to the web version of the vignette where technically those sections will no longer exist. 

Maybe we want to change "you may want to look at one of the following links" to "please refer to the "Developers" section on our pkgdown site", though this is still clunky as we can't link to the right URL.

Reply:
https://arrow.apache.org/docs/r/ is now updated for 7.0.0, so is this change needed?

Reply:
The only 2 complaints left are:
```r
> urlchecker::url_check()
! Warning: vignettes/arrow.Rmd:73:8 Moved
gains](http://arrow.apache.org/blog/2019/01/25/r-spark-improvements/).
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       https://arrow.apache.org/blog/2019/01/25/r-spark-improvements/
x Error: README.md:5:75 400: Bad Request
[![conda-forge](https://img.shields.io/conda/vn/conda-forge/r-arrow.svg)](https://anaconda.org/conda-forge/r-arrow)
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Not sure what the deal is with the conda one - it seems correct. The spark blog one: the URL is an https, not http.

Reply:
Done. Reverted the 2 other changes. Basically this PR will add the letter `s`. 😄